pkg/tsdb/influxdb/response_parser.go

```diff
@@ -56,32 +56,55 @@ func parseJSON(buf io.ReadCloser) (Response, error) {
 func transformRows(rows []Row, query Query) data.Frames {
 	frames := data.Frames{}
 	for _, row := range rows {
-		for columnIndex, column := range row.Columns {
+		var hasTimeCol = false
+
+		for _, column := range row.Columns {
 			if column == "time" {
```

Comment:
nit: Does it guaranteed that the time column always have label 'time"? 🤔

Reply:
It does not! But hasTimeCol is set to false by default, so if it does not, we will take it that this is a metricFindQuery.